### PR TITLE
merge from master and replace some new nose test with pytest

### PR DESCRIPTION
**Pull request by laerus**
atm the only failling test i see in python3.5 is:

```
        for inp, exp in getitem_cases:
            obs = EnvPath(inp)[0]  # call to __getitem__
>           assert expand(exp) == obs
E           assert '/home/laerus/' == '~/'
E             - /home/laerus/
E             + ~/

tests/test_tools.py:572: AssertionError
```
which seems like a legit bug in EnvPath

### CODE REVIEW

File: tests/test_imphooks.py [hunk 1]
```diff
@@ -10,7 +10,9 @@
 from tools import mock_xonsh_env
 LOADED_HERE = False
 
-def setup_module():
+IMP_ENV = {'PATH': [], 'PATHEXT': []}
+
+def setup():
```
Comment (gforsyth) [inline tests/test_imphooks.py]: should this be `setup_module` for pytest?

Comment (laerus) [inline tests/test_imphooks.py]: it was a conflict with the master merge and i kept the master part, will change it

File: tests/tools.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
 VER_MAJOR_MINOR = sys.version_info[:2]
 VER_FULL = sys.version_info[:3]
 ON_MAC = (platform.system() == 'Darwin')
+ON_WINDOWS = (platform.system() == 'Windows')
```
Comment (gforsyth) [inline tests/tools.py]: shoudl we just import this from `xonsh.platform`?  same for `ON_DARWIN` (instead of `ON_MAC`)

Comment (laerus) [inline tests/tools.py]: on it!

**Comment by gforsyth** [comment]
Awesome.  Thanks @laerus !

**Pull request merged**